src/Functions/formatDateTime.cpp

```diff
@@ -126,6 +126,8 @@ constexpr std::string_view monthsFull[]
 constexpr std::string_view monthsShort[]
     = {"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
 
+constexpr std::string_view dynamicDateFormatters[2] = {"M", "W"};
```

Comment:
I guess you want to check against `%M` and `%W`, otherwise this will match the format string `Month: %m Weekday: %w` (which does _not_ contain dynamic formatters)